docs/react-v9/contributing/component-implementation-guide.md

```diff
@@ -0,0 +1,264 @@
+
+## Component Spec
+
+### Resources
+
+- Start with the [Spec Template](Spec-Template)
+- Refer to the [Best Practices](https://github.com/microsoft/fluentui/wiki/Implementation-Best-Practices) while designing the component
+
+Write the spec in a file named `SPEC.md` in the component's folder. The spec should document your research above, and outline the proposed API, structure, and usage of the component.
+
+The spec should include the following:
+
+- Link to the GitHub epic issue
+- Relevant background research
+  - Details about what exists today in Fabric/v8 and Northstar/v0
+  - Link to the Open UI research
+  - Any relevant customer asks based on the open issues
+- The proposed API: the set of props, slots, and tokens for the component.
+  - Ideally you'd write the component's `.types.ts` file and use that directly as the API proposal.
+- A few examples of code using the component. These could take the form of what would eventually be in an example from the documentation.
+- A plan for migration from the v8 and v0 versions of the component.
+
+### Spec Review
+
+The reviewers will help to double check the spec, and the API is consistent with other Fluent UI controls. The review is _not_ intended to check every aspect of the design to ensure that the control works for all use cases--you are still accountable to create a high quality design for the control.
+
+To get a review started:
+
+- Create a pull request that contains `SPEC.md` and `.types.ts` (if applicable).
+  - If you have any open questions, call them out with a comment on the PR.
+  - Assign all of the core spec reviewers. Your PR should get signed off by at least one of these reviewers to be considered complete:
+    - Levi (levithomason)
+    - Miro (miroslavstastny)
+    - Ben (behowell)
+  - Anyone can provide input or feedback on Specs
+- If needed, create a spec review meeting or use Spec Review - Office Hours to walk through the spec, and include a link to the PR in the meeting invite.
+  - If you have access to the Fluent UI Internal Teams channel, post a link to the spec in the Convergence channel so people know to look at the spec. At-mention the reviewers to ensure it's on their radar.
+- The spec does not have to be final with all issues or open questions resolved before merging the first draft.
+  - If the open questions are large enough to warrant their own discussion, consider creating separate issues to track those open questions.
+
+# Phase 2: Initial Implementation
+
+Once the first draft of the spec has been reviewed, you can get started on the implementation.
+
+## Implementation
+
+**Requirements**
+
+- Use React hooks
+- Use makeStyles
+- Respect Figma tokens (and use the Provider)
+- Respect API principles, shorthands and slots handling
+- No dependency on v0/v7/v8
+- Add tsdoc `/** */` comments to all public APIs in your component
```

Comment:
nobody is following this , we dont have any tooling in place to enforce this, might be introduced later but at this stage I'd suggest to remove this